libs/SmartSync/src/com/salesforce/androidsdk/smartsync/manager/CacheManager.java

```diff
@@ -749,22 +749,27 @@ private boolean doesMasterSoupContainSoup(String soupName) {
      * @return List of soup names.
      */
     private JSONArray getAllSoupNames() {
-    	final String smartSql = "SELECT {" + SOUP_OF_SOUPS + ":" +
-        		SOUP_NAMES_KEY + "} FROM {" + SOUP_OF_SOUPS + "}";
         JSONArray results = null;
-    	QuerySpec querySpec = QuerySpec.buildSmartQuerySpec(smartSql, 1);
-        try {
-            int count = smartStore.countQuery(querySpec);
-            querySpec = QuerySpec.buildSmartQuerySpec(smartSql, count);
-			results = smartStore.query(querySpec, 0);
-		} catch (JSONException e) {
-            Log.e(TAG, "JSONException occurred while attempting to read cached data", e);
-		} catch (SmartStoreException e) {
-            Log.e(TAG, "SmartStoreException occurred while attempting to read cached data", e);
+
+        if (smartStore.hasSoup(SOUP_OF_SOUPS)) {
```

Comment:
An exception was showing up in the log after calling hardReset on the CacheManager (which we do in the tear down of smartsync tests) when it had never been used.